test/instances/ezETH/EzETHHyperdrive.t.sol

```diff
@@ -0,0 +1,1110 @@
+        advanceTime(POSITION_DURATION_2_WEEKS, variableRate);
+
+        // Bob attempts to close his short with ETH as the target asset. This
+        // fails since ETH isn't supported as a withdrawal asset.
+        vm.stopPrank();
+        vm.startPrank(bob);
+        vm.expectRevert(IHyperdrive.UnsupportedToken.selector);
+        hyperdrive.closeShort(
+            maturityTime,
+            shortAmount,
+            0,
+            IHyperdrive.Options({
+                destination: bob,
+                asBase: true,
+                extraData: new bytes(0)
+            })
+        );
+    }
+
+    function test_close_short_with_ezeth(
+        uint256 shortAmount,
+        int256 variableRate
+    ) external {
+        // Accrue interest for a term to ensure that the share price is greater
+        // than one.
+        advanceTime(POSITION_DURATION_2_WEEKS, 0.05e18);
+
+        // Bob opens a short.
+        shortAmount = shortAmount.normalizeToRange(
+            2 * hyperdrive.getPoolConfig().minimumTransactionAmount,
+            HyperdriveUtils.calculateMaxShort(hyperdrive)
+        );
+        uint256 balanceBefore = bob.balance;
+        vm.deal(bob, shortAmount);
+        (uint256 maturityTime, uint256 basePaid) = openShort(bob, shortAmount);
+        vm.deal(bob, balanceBefore - basePaid);
+
+        // The term passes and interest accrues.
+        uint256 startingVaultSharePrice = hyperdrive
+            .getPoolInfo()
+            .vaultSharePrice;
+        variableRate = variableRate.normalizeToRange(0, 2.5e18);
+        advanceTime(POSITION_DURATION_2_WEEKS, variableRate);
+
+        // Get some balance information before closing the short.
+        (
+            ,
+            uint256 totalPooledEtherBefore,
+            uint256 totalSharesBefore
+        ) = getSharePrice();
+        AccountBalances memory bobBalancesBefore = getAccountBalances(bob);
+        AccountBalances memory hyperdriveBalancesBefore = getAccountBalances(
+            address(hyperdrive)
+        );
+
+        // Bob closes his short with ezETH as the target asset. Bob's proceeds
+        // should be the variable interest that accrued on the shorted bonds.
+        uint256 expectedBaseProceeds = shortAmount.mulDivDown(
+            hyperdrive.getPoolInfo().vaultSharePrice - startingVaultSharePrice,
+            startingVaultSharePrice
+        );
+        uint256 shareProceeds = closeShort(
+            bob,
+            maturityTime,
+            shortAmount,
+            false
+        );
+        (, uint256 totalPooledEther, uint256 totalShares) = getSharePrice();
+        uint256 baseProceeds = shareProceeds.mulDivDown(
+            totalPooledEther,
+            totalShares
+        );
+        // TODO: make sure this is OK.  vaultSharePrice is pretty inaccurate.
+        assertLe(baseProceeds, expectedBaseProceeds + 1e4);
+        assertApproxEqAbs(baseProceeds, expectedBaseProceeds, 1e5); // was 100
```

Comment:
I'm not sure if this is okay. These tolerances seem pretty high, but it feels unavoidable because of the way that they do their share price calculations: https://github.com/Renzo-Protocol/contracts-public/blob/0530cbd20590677da5fa6fd32a0a8b250a9eb028/contracts/Oracle/RenzoOracle.sol#L111. 

I spent some time seeing if I could eliminate the tolerances, and I couldn't because their share price logic is inherently imprecise. I'm not sure why these chose this calculation, but it feels like we're stuck with it. 

With all this said, I think we should do some more rigorous testing of this yield source over the course of the audit. I'm fine with removing the TODOs and merging this for now though.

Reply:
Ok sounds good to me, lemme know if you wanna sync up to figure out how to do that.